### ISSUE TITLE
Checksum fails for telegram-2.15-47337.zip

### ISSUE THREAD

**Issue by chiselwright** [caskroom/homebrew-cask#22270]
Trying to install/upgrade telegram I'm seeing:

```
➔ brew update && brew cask install telegram
Already up-to-date.
==> Downloading https://osx.telegram.org/updates/Telegram-2.15-47337.app.zip
Already downloaded: /Users/c.wright/Library/Caches/Homebrew/telegram-2.15-47337.zip
==> Verifying checksum for Cask telegram
==> Note: running "brew update" may fix sha256 checksum errors
Error: sha256 mismatch
Expected: 53166186b69d789f94d9153963ff9b20791a83d9e3f73ae644e8772ae911b234
Actual: 89643b017b560ec177ac306ded6996def9db8e920165198c09993165ac75c010
File: /Users/c.wright/Library/Caches/Homebrew/telegram-2.15-47337.zip
To retry an incomplete download, remove the file above.
```

I *have* removed the file as recommended. No change,

It seems slightly odd that the version of telegram jumped from 0.9.51 to 2.15-47337.

I believe the checksum is wrong in the telegram.rb file:
```
➔ curl -s https://osx.telegram.org/updates/Telegram-2.15-47337.app.zip |shasum --algorithm 256
89643b017b560ec177ac306ded6996def9db8e920165198c09993165ac75c010  -
```

I'll submit a pull-request shortly to address this.

On a side note:

Some further investigation indicates there's been a change from "telegram" to "telegram desktop", without any real warning to those already using telegram:
```
086b8fa telegram.rb: updated vendor comment
23823ce updated telegram (2.15-47337) (#22154)
579dc5d Adds Telegram for macOS and renames Telegram Desktop. (#22092)
392d0af updated telegram (0.9.51) (#21998)
b942523 updated telegram (0.9.50) (#21997)
2ba2b73 updated telegram (0.9.49) (#21235)
370d554 updated telegram (0.9.48) (#21116)
```

Info on the two casks:
```
➔ brew cask info telegram
telegram: 2.15-47337
Telegram for macOS
https://macos.telegram.org
Not installed
https://github.com/caskroom/homebrew-cask/blob/master/Casks/telegram.rb
==> Contents
  Telegram.app (app)

➔ brew cask info telegram-desktop
telegram-desktop: 0.9.51
Telegram Desktop
https://desktop.telegram.org
Not installed
https://github.com/caskroom/homebrew-cask/blob/master/Casks/telegram-desktop.rb
==> Contents
  Telegram.app (app)
```

**Comment by chiselwright** [comment]
Pull request here: https://github.com/caskroom/homebrew-cask/pull/22271

**Comment by fanquake** [comment]
Fixed in #22267
